api/bucketsd/service.go

```diff
@@ -2337,17 +2694,40 @@ func (s *Service) RemovePath(ctx context.Context, req *pb.RemovePathRequest) (re
 	defer lck.Release()
 
 	buck := &tdb.Bucket{}
-	err = s.Buckets.GetSafe(ctx, dbID, req.Key, buck, tdb.WithToken(dbToken))
-	if err != nil {
+	if err := s.Buckets.GetSafe(ctx, dbID, req.Key, buck, tdb.WithToken(dbToken)); err != nil {
 		return nil, err
 	}
+
 	if req.Root != "" && req.Root != buck.Path {
 		return nil, status.Error(codes.FailedPrecondition, buckets.ErrNonFastForward.Error())
 	}
 
+	dirPath, err := s.removePath(ctx, dbID, dbToken, buck, filePath)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
```
buck.UpdatedAt = time.Now().UnixNano()
buck.UnsetMetadataWithPrefix(filePath)
```
^ needs to happen before Verify is called. Otherwise, we're not verifying anything.

Reply:
ahhh, thanks. i misunderstood the role of that. i think it doesn't make sense in the `removePath` but should be called first before any method that calls that.

Reply:
done